atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RangeLoader.java

```diff
@@ -65,20 +69,38 @@ public void setConsistencyLevel(ConsistencyLevel consistencyLevel) {
             ConsistencyLevel consistency,
             Supplier<ResultsExtractor<T>> resultsExtractor) {
         SlicePredicate predicate;
+
         if (rangeRequest.getColumnNames().size() == 1) {
-            byte[] colName = rangeRequest.getColumnNames().iterator().next();
-            predicate = SlicePredicates.latestVersionForColumn(colName, startTs);
+            predicate = getSlicePredicateForSingleColumn(rangeRequest, startTs);
         } else {
-            // TODO(nziebart): optimize fetching multiple columns by performing a parallel range request for
-            // each column. note that if no columns are specified, it's a special case that means all columns
-            predicate = SlicePredicates.create(SlicePredicates.Range.ALL, SlicePredicates.Limit.NO_LIMIT);
+            TableMetadata tableMetadata = metaDataProvider.getMetaDataFromRef(tableRef);
+            // todo (sudiksha) : add null check, also refactor
+            Set<byte[]> allColumnNames = tableMetadata.getColumns().getAllColumnNames();
+            if (allColumnNames.size() == 1) {
+                RangeRequest newRangeRequest = RangeRequest.builder().startRowInclusive(rangeRequest.getStartInclusive())
```

Comment:
non-actionable as it might actually be worse, would it make sense to basically call this method but with the range request you just created? @jeremyk-91 what do you think?

Reply:
That said, see James's comment: this rewriting should be done at the codegen layer, in which case you'll probably be calling transactional/KVS methods with the different range request in any case